### ISSUE TITLE
[proposal] tooltip: change root element from div to span

### ISSUE THREAD

**Issue by pimlie** [bootstrap-vue/bootstrap-vue#571]
I am trying to give some (inline) images a tooltip but because of the div tag the images are not inline anymore after applying the tooltip.

In vanilla bootstrap you set the tooltip attributes directly on the element that you want to have the tooltip, so the tooltip component should not interfere with the page layout by wrapping everything in a block element. If you agree I can make a PR for this, but its such a small change maybe its quicker you do it :)

**Comment by tmorehouse** [comment]
a true block-level `div` might not be best (and `span` isn't ideal since, technically, block level elements aren't allowed in `span`s).

An in-line block would probably be best (under the current implementation), Maybe a div element with `style="display:inline-block;"`, as block level elements typically span 100% width and would definitely break some layouts.

But please note that tooltips and popovers will be changing significantly when Bootstrap V4.beta.1 is released (they are replacing tether.js with popper.js), and the implementation in Bootstrap-vue will also be changing significantly when this happens.

**Comment by tmorehouse** [comment]
I just realized that you re referring to the wrapper around the tooltip itself, and not the wrapper that the component places around the original triggering element.

**Comment by tmorehouse** [comment]
Just to clarify, re you referring to the wrapper for the tooltip element itself, or the wrapper that wraps the trigger element (and the tooltip)?

**Comment by pimlie** [comment]
The latter, so the <div> on line 2 and line 15, the div that's only there because a template can have only one root element :)

**Comment by tmorehouse** [comment]
As a test, could you try adding `class="d-inline-block"` to your `b-tooltip` component, and see if that fixes the issue?

This class should be directly applied to the outer root div.

**Comment by pimlie** [comment]
```
<b-tooltip content="hi" class="b-inline-block">
  <img src="https://bootstrap-vue.github.io/_nuxt/img/banner.f0a8c9e.png"/>
</b-tooltip>
```
renders indeed as as
```
<div class="b-inline-block">
  <span>
    <img src="https://bootstrap-vue.github.io/_nuxt/img/banner.f0a8c9e.png" class="">
  </span>
</div>
```

**Comment by tmorehouse** [comment]
And does your layout work correctly?

**Comment by pimlie** [comment]
Yes, it works. If you try in the playground with and without the d-inline-block class the images render side-by-side and below each other respectedly

**Comment by tmorehouse** [comment]
fixed in v0.17.1
